backend/eventd/eventd.go

```diff
@@ -246,9 +246,12 @@ func (e *Eventd) handleMessage(msg interface{}) error {
 
 	ctx := context.WithValue(context.Background(), corev2.NamespaceKey, event.Entity.Namespace)
 
-	// Create a proxy entity if required and update the event's entity with it
-	if err := createProxyEntity(event, e.store); err != nil {
-		return err
+	// Create a proxy entity if required and update the event's entity with it,
+	// but only if the event's entity is not an agent.
+	if event.Entity.EntityClass != corev2.EntityAgentClass {
```

Comment:
That won't work for check with `proxy_entity_name` since the agent entity is returned as the event's entity!